spec/lexers/kotlin_spec.rb

```diff
@@ -22,5 +22,77 @@
     it 'recognizes one-line comments not followed by a newline (#797)' do
       assert_tokens_equal '// comment', ['Comment.Single', '// comment']
     end
+
+    describe 'functions' do
+
+      it 'recognizes function' do
+        assert_tokens_equal 'fun makeField()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes backticked function' do
+        assert_tokens_equal 'fun `makeField`()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Function', '`makeField`'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes extension function' do
+        assert_tokens_equal 'fun String.makeField()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Class', 'String'],
+          ['Punctuation', '.'],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes backticked extension function' do
+        assert_tokens_equal 'fun `String`.`makeField`()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Name.Class', '`String`'],
+          ['Punctuation', '.'],
+          ['Name.Function', '`makeField`'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes generic function' do
+        assert_tokens_equal 'fun <T, ClassT> makeField()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Punctuation', '<'],
+          ['Name.Class', 'T'],
+          ['Punctuation', ','],
+          ['Text', " "],
+          ['Name.Class', 'ClassT'],
+          ['Punctuation', '>'],
+          ['Text', " "],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '()']
+      end
+
+      it 'recognizes generic extension function' do
+        assert_tokens_equal 'fun <T, ClassT> String.makeField()',
+          ['Keyword', 'fun'],
+          ['Text', " "],
+          ['Punctuation', '<'],
+          ['Name.Class', 'T'],
+          ['Punctuation', ','],
+          ['Text', " "],
+          ['Name.Class', 'ClassT'],
+          ['Punctuation', '>'],
+          ['Text', " "],
+          ['Name.Class', 'String'],
+          ['Punctuation', '.'],
+          ['Name.Function', 'makeField'],
+          ['Punctuation', '()']
+      end
+
+    end
```

Comment:
Same here as in #1001. Please remove these, but thank you for the sentiment. If there aren't already examples covering this in samples, please add those as well.

Reply:
Tests removed and adding missing samples.